source/implementations/f7/Meadow.F7/Devices/F7GPIOManager_interrupts.cs

```diff
@@ -188,37 +189,40 @@ private void InterruptServiceThreadProc(object o)
             if (_firstInterrupt)
             {
                 // DEV NOTE: this is to force the interp pipeline top build at least some of the call stack and improve the response of first-interrupt results
-                Interrupt?.Invoke(new NullPin(), false);
+                Interrupt?.Invoke(_nullPin, false);
                 _firstInterrupt = false;
             }
 
             int priority = 0;
-            try
+
+            var result = Interop.Nuttx.mq_receive(queue, rx_buffer, rx_buffer.Length, ref priority);
+
+            // byte 1 contains the port and pin, byte 2 contains the stable state.
+            if (result >= 0)
             {
-                var result = Interop.Nuttx.mq_receive(queue, rx_buffer, rx_buffer.Length, ref priority);
+                var irq = rx_buffer[0];
+                bool state = rx_buffer[1] != 0;
+                var port = irq >> 4;
+                var pin = irq & 0xf;
+                IPin ipin;
+
 
-                // byte 1 contains the port and pin, byte 2 contains the stable state.
-                if (result >= 0)
+                lock (_interruptPins)   //FIXME: If this is supposed to keep modifications from
```

Comment:
The intent of this lock, I think, was to prevent the IST from trying to get a pin from the collection at the same time an interrupt was potentially removed via a Dispose on an InterruptPort

Reply:
It won't do that currently. `lock` takes a mutex using the _interruptPins object as a handle. A second thread would have to contend with this mutex in order for this to cause any synchronization. Since there is only one lock statement and one IST thread (I think), this is sort of a NOP currently.